### PR TITLE
Reposition on window resize

### PR DESCRIPTION
**Pull request by spencerfdavis**
This pull request closes https://github.com/rpearce/react-medium-image-zoom/issues/162.

### CODE REVIEW

File: source/Activated.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React, { useCallback, useEffect, useRef, useState } from 'react'
 import { bool, func, node, object, string } from 'prop-types'
+import { useWindowSize } from './helpers'
```
Comment (spencerfdavis) [inline source/Activated.js]: How do you feel about having this hook in the helpers?

File: source/helpers.js [hunk 2]
```diff
@@ -22,6 +24,27 @@ export const unzoom = id => {
...
+const getSize = () => ({
+  innerHeight: window.innerHeight,
+  innerWidth: window.innerWidth,
+  outerHeight: window.outerHeight,
+  outerWidth: window.outerWidth
+})
+
+export const useWindowSize = () => {
+  const [windowSize, setWindowSize] = useState(getSize())
+  const handleResize = () => setWindowSize(getSize())
```
Comment (spencerfdavis) [inline source/helpers.js]: Do you think this needs to be throttled?

Comment (rpearce) [reply source/helpers.js]: We weren't doing it before (https://github.com/rpearce/react-medium-image-zoom/blob/v3.x/src/EventsWrapper.js#L15)... I say we throttle when we need to.

File: source/Activated.js (hunk 1 above)
Comment (rpearce) [reply source/Activated.js]: fine by me. happy to add things there until it gets unwieldy

File: source/helpers.js (hunk 2 above)
Comment (rpearce) [reply source/helpers.js]: I actually just realized that this helpers file is going away (not using anything in it).

I'd move this to a different file.

Should we use https://github.com/streamich/react-use/blob/master/docs/useWindowSize.md, or do we need the different implementation?

**Comment by rpearce** [comment]
This looks prettay good :fire:

If we get https://github.com/rpearce/react-medium-image-zoom/pull/171 in, we can add a test here, but if not, we can add the test after

File: source/helpers.js (hunk 2 above)
Comment (spencerfdavis) [reply source/helpers.js]: Good suggestion.  Nothing special.  Let's use the `react-use` one.

File: source/Activated.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import React, { useCallback, useEffect, useRef, useState } from 'react'
 import { bool, func, node, object, string } from 'prop-types'
-import { useWindowSize } from './helpers'
+import { useWindowSize } from 'react-use'
```
Comment (rpearce) [inline source/Activated.js]: Per the [Usage docs](https://github.com/streamich/react-use/blob/master/docs/Usage.md), maybe should do it like
```js
import useWindowSize from 'react-use/lib/useWindowSize'
```
so we only pull that thing in?

File: source/Activated.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
-import React, { useCallback, useEffect, useRef, useState } from 'react'
 import { bool, func, node, object, string } from 'prop-types'
+import React, { useCallback, useEffect, useRef, useState } from 'react'
```
Comment (rpearce) [inline source/Activated.js]: FYI -- this is annoying, but I think we always want to keep `react` as the first import no matter what. There was something back in the day recommending this b/c of different versions of react being run if a dependency also requires react...

Comment (rpearce) [reply source/Activated.js]: once you move this to the top, merge away

**Comment by spencerfdavis** [comment]
image

**Pull request merged**